_core/smart/overview.md

```diff
@@ -1,135 +1,55 @@
-
-For the pre-release, the Blockstack test environment includes the `clarity-cli` command for interacting with the contract and SQLite to support the data space. You create a SQLLite database to hold data related to Clarity contracts. This database simulates the blockchain by recording the contract activity. 
-
-You can't run even an a hello-world program without first initializing a Clarity contract's data space within the database. You can use the `clarity-cli initialize` command to set up the database.
-
-```clarity-cli initialize /data/db```
-
-This command initializes the `db` database which resides in the `/data` directory of the container. You can name the database anything you like, the name `db` is not required. You can use SQLite to query this database:
-
-```sql
-sqlite> .open db
-sqlite> .tables
-contracts            maps_table           type_analysis_table
-data_table           simmed_block_table 
-sqlite> 
-```
-
-After you initialize the contract's data space, you can `check` a Clarity program for problems.
-
-```clarity-cli check ./hello.clar /data/db```
-
-As the name implies, the `check` ensures the contract definition passes a type check; passing programs will returns an exit code of `0` (zero). Once a contract passes a check, you `launch` it.
-
-```bash
-root@4224dd95b5f5:/data# clarity-cli launch hello ./hello.clar /data/db
-Buffer(BuffData { data: [104, 101, 108, 108, 111, 32, 119, 111, 114, 108, 100] })
-Contract initialized!
-```
-
-Because Clarity does not support simple strings, it stores the `hello world` string in a buffer. Printing out that string displays the ASCII representation for each character. You can see the record of this contract's launch in the corresponding database:
+* The language is not intended to be compiled
+* The language is not Turing complete
+  
+These differences allow for static analysis of programs to determine properties like runtime cost and data usage. Omitting compilation prevents the possibility of error or bugs introduced at the compiler level.
 
-```sql
-sqlite> select * from contracts;
-1|hello|{"contract_context":{"name":"hello","variables":{},"functions":{}}}
-sqlite> select * from type_analysis_table;
-1|hello|{"private_function_types":{},"variable_types":{},"public_function_types":{},"read_only_function_types":{},"map_types":{}}
-sqlite> 
-```
+A Clarity smart contract is composed of two parts &mdash; a data-space and a set of functions. Only the associated smart contract may modify its corresponding data-space on the Blockchain. Functions are private unless they are defined as public functions. Users call smart contracts' public functions by broadcasting a transaction on the Blockchain which invokes the public function.
 
-## Language rules and limitations
+Contracts can also call public functions from other smart contracts. The ability to do a static analysis of a smart contract allows a user to determine dependency between contracts.
```

Comment:
This isn't due to anything special in Clarity -- you can determine a transaction's call graph in Solidity as well (albeit doing so requires _running_ the code).

I think this is meant to read something like "The fact that Clarity is a decidable language means that users can determine the set of _reachable_ code from any function they want to call."